### PR TITLE
hypershift: add ovnkube-node-proxy container in ovnkube-node ds 

### PR DESCRIPTION
**Pull request by zshi-redhat**

### CODE REVIEW

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml [hunk 1]
```diff
@@ -83,6 +97,10 @@ spec:
             fi
             sleep 2
           done
+
+          {{if .ENABLE_OVN_NODE_PROXY}}
```
Comment (alvaroaleman) [inline bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:101]: Genuine question, does the container runtime not do that (kill all processes of a container when the container ends)?

**Comment by alvaroaleman** [comment]
Just tried this out, seems to work fine! I would probably squash the last two commits, I looked at the first and got confused why there is no new DS, until I saw that you added a new container to the existing one in the second commit :upside_down_face:

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml [hunk 2]
```diff
@@ -56,6 +56,12 @@ spec:
       hostNetwork: true
       hostPID: true
       priorityClassName: "system-node-critical"
+      {{if .ENABLE_OVN_NODE_PROXY}}
```
Comment (squeed) [inline bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:59]: A comment here explaining what this does would be good :)

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml [hunk 3]
```diff
@@ -72,8 +78,16 @@ spec:
...
+          {{if .ENABLE_OVN_NODE_PROXY}}
+          # Don't use /etc/hosts to resolve OVN_SB_DB_ROUTE_HOST which points to 127.0.0.1 (see spec.hostAliases)
+          echo "hosts: dns files" >> /etc/nsswitch.conf
+          exec socat TCP-LISTEN:{{.OVN_SB_DB_ROUTE_LOCAL_PORT}},reuseaddr,fork PROXY:{{.HTTP_PROXY_IP}}:{{.OVN_SB_DB_ROUTE_HOST}}:{{.OVN_SB_DB_ROUTE_PORT}},proxyport={{.HTTP_PROXY_PORT}} &
+          proxypid=$!
+         {{end}}
+
           ovndb_ctl_ssl_opts="-p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt"
-          sbdb_ip="{{.OVN_SB_DB_ENDPOINT}}"
+          sbdb_ip="{{.OVN_NB_DB_ENDPOINT}}"
```
Comment (squeed) [inline bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:90]: OVN_NB?

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml (hunk 1 above)
Comment (squeed) [reply bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:101]: Not always. Being pid1 is special. I'm not sure if that is the case here, but there are definitely weird issues.

Specifically, pid1 doesn't get signals unless explicitly declared.

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml [hunk 4]
```diff
@@ -100,6 +118,38 @@ spec:
...
+            fieldRef:
+              fieldPath: spec.nodeName
+        volumeMounts:
+        - mountPath: /env
+          name: env-overrides
+        terminationMessagePolicy: FallbackToLogsOnError
+        resources:
+          requests:
+            cpu: 10m
+            memory: 300Mi
```
Comment (squeed) [inline bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:151]: This seems like a lot -- do we need 300 MiB? Can we kick this down a bit?

File: pkg/controller/operconfig/mtu_probe.go [hunk 5]
```diff
@@ -134,13 +144,16 @@ func (r *ReconcileOperConfig) deleteMTUProber(ctx context.Context, infra *bootst
 	return nil
 }
 
-func renderMTUProber(infra *bootstrap.InfraStatus) ([]*uns.Unstructured, error) {
+func renderMTUProber(infra *bootstrap.InfraStatus, proxyStatus configv1.ProxyStatus) ([]*uns.Unstructured, error) {
```
Comment (squeed) [inline pkg/controller/operconfig/mtu_probe.go:147]: nit: make proxyStatus part of the bootstrap.InfraStatus, don't add it as another function argument.

File: pkg/controller/operconfig/operconfig_controller.go [hunk 6]
```diff
@@ -5,10 +5,11 @@ import (
 	"fmt"
 	"log"
 	"reflect"
-	"sigs.k8s.io/controller-runtime/pkg/cluster"
 	"strings"
 	"time"
 
+	"sigs.k8s.io/controller-runtime/pkg/cluster"
```
Comment (squeed) [inline pkg/controller/operconfig/operconfig_controller.go:11]: remove this change, I fixed it in another PR :)

**Comment by squeed** [comment]
a few minor nits, but I think this looks good! Thanks!

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml (hunk 2 above)
Comment (zshi-redhat) [reply bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml:64]: Done

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml (hunk 3 above)
Comment (zshi-redhat) [reply bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml]: Fixed

File: bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml (hunk 4 above)
Comment (zshi-redhat) [reply bindata/network/ovn-kubernetes/managed/ovnkube-node.yaml]: Fixed

File: pkg/controller/operconfig/mtu_probe.go (hunk 5 above)
Comment (zshi-redhat) [reply pkg/controller/operconfig/mtu_probe.go]: Fixed

File: pkg/controller/operconfig/operconfig_controller.go (hunk 6 above)
Comment (zshi-redhat) [reply pkg/controller/operconfig/operconfig_controller.go:11]: I rebased against master, this line still exists. 
Could you link me to the PR that fixed this?

**Comment by zshi-redhat** [comment]
> Just tried this out, seems to work fine! I would probably squash the last two commits, I looked at the first and got confused why there is no new DS, until I saw that you added a new container to the existing one in the second commit upside_down_face

Squashed the last two commits. I first added ovnkube-node-proxy as a separate daemonset, then changed to use existing ds after talking with @kyrtapz .

**Pull request merged**